atlassian_jwt_auth/contrib/django_app/auth.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+import atlassian_jwt_auth
+import jwt
+import requests
+
+from django.conf import settings
+
+from .models import Issuer
+
+
+logger = logging.getLogger(__name__)
+
+
+class ASAPAuthBackend():
+    def authenticate(token):
+        verifier = _get_verifier()
+        allowed_aud = settings.ASAP_ALLOWED_AUDIENCE
+
+        try:
+            claims = verifier.verify_jwt(
+                token, allowed_aud, leeway=60
+            )
+        except (
+            requests.exceptions.HTTPError,
+            requests.exceptions.ConnectionError,
+            jwt.Exceptions.InvalidTokenError
+        ):
+            logging.exception('Authentication failed while verifying ASAP token')
+            return None
+
+        try:
+            return Issuer.objects.get(issuer=claims['iss']).user
+        except Issuer.DoesNotExist:
+            return None
+
+
+def _get_verifier():
+    """Returns a verifier based on config.asap['PUBLICKEY_REPOSITORY']"""
+    return atlassian_jwt_auth.JWTAuthVerifier(
+        atlassian_jwt_auth.HTTPSPublicKeyRetriever(
+            settings.ASAP_PUBLICKEY_REPOSITORY
```

Comment:
Please use `getattr(settings, 'ASAP_PUBLICKEY_REPOSITORY', None)` as `settings.ASAP_PUBLICKEY_REPOSITORY` may not be set.